### PR TITLE
[SAMPLES] Add GNA library version log to speech_sample

### PR DESCRIPTION
**Pull request by kbruniec**
### Details:
 - Add GNA library version log to speech_sample
 - Add OV version (eg hash commit) to TLV model file

### Tickets:
 - 86924

### CODE REVIEW

**Comment by marcinkusm** [comment]
It would be good to add all information about introduced changes  to commit messages and PR description. Currently it reflects only part of the changes.

File: samples/cpp/speech_sample/main.cpp [hunk 1]
```diff
@@ -85,6 +85,12 @@ int main(int argc, char* argv[]) {
         // --------------------------- Step 1. Initialize OpenVINO Runtime core and read model
         // -------------------------------------
         ov::Core core;
+        try {
+            const auto& gnaLibraryVersion = core.get_property("GNA", ov::intel_gna::library_full_version);
+            slog::info << "Detected GNA Library: " << gnaLibraryVersion << slog::endl;
+        } catch (std::exception& e) {
+            slog::info << "Can not detect GNA Library version, exception: " << e.what() << slog::endl;
```
Comment (ilya-lavrenov) [inline samples/cpp/speech_sample/main.cpp:92]: why can we fail to detect GNA version?

Comment (kbruniec) [reply samples/cpp/speech_sample/main.cpp:92]: > why can we fail to detect GNA version?
It should not, but I wanted to make it more safe for potential (future?) bugs.
Do you think making it more safe is big problem and I should remove try block?

Comment (kbruniec) [reply samples/cpp/speech_sample/main.cpp:92]: I think one potential problem may be on integration side - the GNA Plugin may not load (eg., due to dll not found), but we could still use speech_sample with CPU plugin.

File: src/plugins/intel_gna/common/versioning.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+//
+
+#include "versioning.hpp"
+
+#include <sstream>
+#include <string>
+
+#include "openvino/core/version.hpp"
+
+namespace GNAPluginNS {
```
Comment (nkogteva) [inline src/plugins/intel_gna/common/versioning.cpp:12]: Let's use `ov::intel_gna` namespace for new code

File: src/plugins/intel_gna/common/versioning.cpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+namespace common {
+
+std::string GetVersionOfOv() {
```
Comment (nkogteva) [inline src/plugins/intel_gna/common/versioning.cpp:15]:
```suggestion
std::string get_openvino_version_string() {
```

Comment (kbruniec) [reply src/plugins/intel_gna/common/versioning.cpp:15]: Is there some specific requirement for function naming,
I thought we use "Google C++ Style Guide"?

According to https://google.github.io/styleguide/cppguide.html#Function_Names
"Regular functions have **mixed case**; accessors and mutators may be named like variables."

File: src/plugins/intel_gna/common/versioning.cpp (hunk 2 above)
Comment (kbruniec) [reply src/plugins/intel_gna/common/versioning.cpp:12]: OK

File: src/plugins/intel_gna/common/versioning.cpp (hunk 3 above)
Comment (kbruniec) [reply src/plugins/intel_gna/common/versioning.cpp:15]: @nkogteva please clarify regarding the style

File: samples/cpp/speech_sample/main.cpp (hunk 1 above)
Comment (kbruniec) [reply samples/cpp/speech_sample/main.cpp:92]: @ilya-lavrenov can we resolve this issue?

File: src/plugins/intel_gna/common/versioning.cpp (hunk 3 above)
Comment (nkogteva) [reply src/plugins/intel_gna/common/versioning.cpp:15]: @kbruniec As far I as know, openvino new code uses snake_case, but we still have a lot of code with CamelCase. I'm trying to use new code style in new code. But if you have some doubts, you can keep it as is? we already have a mess. Code style aligning can be done in separate PR later.

**Review by sirzabek — APPROVED** [review]

**Review by nkogteva — APPROVED** [review]

**Pull request merged**